apps/prairielearn/src/pages/instructorGradebook/components/InstructorGradebookTable.tsx

```diff
@@ -0,0 +1,544 @@
+      const columnId = `a${assessment.assessment_id}`;
+      assessmentFilters[columnId] = ({ header }: { header: Header<GradebookRow, unknown> }) => {
+        return <NumericInputColumnFilter column={header.column} />;
+      };
+    });
+
+    return {
+      role: ({ header }: { header: Header<GradebookRow, GradebookRow['role']> }) => (
+        <CategoricalColumnFilter
+          column={header.column}
+          allColumnValues={ROLE_VALUES}
+          renderValueLabel={({ value }) => <span>{value}</span>}
+        />
+      ),
+      enrollment_status: ({ header }: { header: Header<GradebookRow, EnumEnrollmentStatus> }) => (
+        <CategoricalColumnFilter
+          column={header.column}
+          allColumnValues={STATUS_VALUES}
+          renderValueLabel={({ value }) => <EnrollmentStatusIcon type="text" status={value} />}
+        />
+      ),
+      ...assessmentFilters,
+    };
+  }, [courseAssessments]);
+
+  const table = useReactTable({
+    data: gradebookRows,
+    columns,
+    columnResizeMode: 'onChange',
+    getRowId: (row) => row.user_id,
+    state: {
+      sorting,
+      columnFilters,
+      globalFilter,
+      columnSizing,
+      columnVisibility,
+      columnPinning,
+    },
+    initialState: {
+      columnPinning: DEFAULT_PINNING,
+      columnVisibility: defaultColumnVisibility,
+    },
+    onSortingChange: setSorting,
+    onGlobalFilterChange: setGlobalFilter,
+    onColumnSizingChange: setColumnSizing,
+    onColumnFiltersChange: handleColumnFiltersChange,
+    onColumnVisibilityChange: setColumnVisibility,
+    onColumnPinningChange: setColumnPinning,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    getFilteredRowModel: getFilteredRowModel(),
+    defaultColumn: {
+      size: 150,
+      maxSize: 500,
+      enableSorting: true,
+      enableHiding: true,
+      enablePinning: true,
+    },
+  });
+
+  return (
+    <>
+      <TanstackTableCard
+        table={table}
+        title="Gradebook"
+        singularLabel="user"
+        pluralLabel="users"
+        // eslint-disable-next-line @eslint-react/no-forbidden-props
+        className="h-100"
+        headerButtons={
+          <button
+            type="button"
+            class="btn btn-sm btn-light"
+            onClick={() => {
+              window.location.href = `${urlPrefix}/instance_admin/gradebook/${csvFilename}`;
+            }}
+          >
+            <i class="fa fa-download" aria-hidden="true" /> Download
+          </button>
+        }
+        columnManagerButtons={
+          <PresetFilterDropdown
+            table={table}
+            label="Filter"
+            options={{
+              'Joined Students': [
+                { id: 'enrollment_status', value: ['joined'] },
+                { id: 'role', value: ['Student'] },
+              ],
+              Students: [{ id: 'role', value: ['Student'] }],
+              'Students & Staff': [],
```

Comment:
I'm inclined to name these "All Students" and "All Students & Staff" to more clearly indicate how they differ from "Joined Students".

Reply:
Fixed in [d1e1dbe](https://github.com/PrairieLearn/PrairieLearn/pull/13179/commits/d1e1dbe2b57f65bed572522d07c9ef7d3e885cbc)